specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1007 @@
+
+If multiple Subscriptions with the same destination are associated with the same
+Channel, each Subscription MUST generate one delivery attempt per Subscription.
+The implementation MAY attach additional event attributes or other metadata
+distinguishing between these deliveries. The implementation MUST NOT modify the
+event payload in this process.
+
+### Event Delivery
+
+Once a Trigger or Subscription has decided to deliver an event, it MUST do the
+following:
+
+1. Resolve all URLs and delivery options, using the values in `status` for URL
+   resolution.
+
+1. Attempt delivery to the `status.subscriberUri` URL following the
+   [data plane contract](./data-plane.md).
+
+   1. If the event delivery fails with a retryable error, it MUST be retried up
+      to `retry` times (subject to congestion control), following the
+      `backoffPolicy` and `backoffDelay` parameters if specified.
+
+1. If the delivery attempt is successful (either the original request or a
+   retry) and no event is returned, the event delivery is complete.
+
+1. If the delivery attempt is successful (either the original request or a
+   retry) and an event is returned in the reply, the reply event will be
+   delivered to the `status.replyUri` destination (for Subscriptions) or added
+   to the Broker for processing (for Triggers). If `status.replyUri` is not
+   present in the Subscription, the reply event MUST be dropped.
+
+   1. For Subscriptions, if delivery of the reply event fails with a retryable
+      error, the entire delivery of the event to MUST be retried up to `retry`
+      times (subject to congestion control), following the `backoffPolicy` and
+      `backoffDelay` parameters if specified.
+
+1. If all retries are exhausted for either the original delivery or the retry,
+   or if a non-retryable error is received, the event MUST be delivered to the
+   `deadLetterSink` in the delivery options. If no `deadLetterSink` is
+   specified, the event is dropped.
+
+   The implementation MAY set additional attributes on the event or wrap the
+   failed event in a "failed delivery" event; this behavior is not (currently)
+   standardized.
+
+   If delivery of the reply event fails with a retryable error, the delivery to
+   the `deadLetterSink` SHOULD be retried up to `retry` times, following the
+   `backoffPolicy` and `backoffDelay` parameters if specified. Alternatively,
+   implementations MAY use an equivalent internal mechanism for delivery (for
+   example, if the `ref` form of `deadLetterSink` points to a compatible
+   implementation).
+
+## Detailed Resources
+
+The following schema defines a set of REQUIRED or RECOMMENDED resource fields on
+the Knative resource types. Whether a field is REQUIRED or RECOMMENDED is
+denoted in the "Schema Requirement" column. Additional `spec` and `status`
+fields MAY be provided by particular implementations, however it is expected
+that most extension will be accomplished via the `metadata.labels` and
```

Comment:
s/extension/extensions/  - add 's'

Reply:
ok you forced me to actually look :-)   https://www.dictionary.com/browse/extension  && https://www.merriam-webster.com/dictionary/extension
I don't see it listed as a "verb". I even spent a good few minute trying really hard to parse/grok it as a verb.... I just can't do it   LOL

Reply:
Adjusted, good to know. I was looking at: "the action of extending", but "the action" makes it a noun.